docbrown/core/src/tgraph.rs

```diff
@@ -357,102 +465,337 @@ impl TemporalGraph {
+    ) -> Box<dyn Iterator<Item = EdgeView> + Send + '_> {
         let v_pid = self.logical_to_physical[&v];
 
         match d {
-            Direction::OUT => Box::new(self.neighbours_iter_window_t(v_pid, r, d).map(
-                move |(v, t, e_meta)| EdgeView {
+            Direction::OUT => Box::new(self.edges_iter_window_t(v_pid, w, d).map(
+                move |(dst, t, e_meta)| EdgeView {
+                    src_g_id: v,
+                    dst_g_id: self.v_g_id(dst, e_meta),
                     src_id: v_pid,
-                    dst_id: v,
+                    dst_id: dst,
                     t: Some(t),
-                    g: self,
                     e_meta,
                 },
             )),
-            Direction::IN => Box::new(self.neighbours_iter_window_t(v_pid, r, d).map(
-                move |(v, t, e_meta)| EdgeView {
-                    src_id: v,
+            Direction::IN => Box::new(self.edges_iter_window_t(v_pid, w, d).map(
+                move |(dst, t, e_meta)| EdgeView {
+                    src_g_id: self.v_g_id(dst, e_meta),
+                    dst_g_id: v,
+                    src_id: dst,
                     dst_id: v_pid,
                     t: Some(t),
-                    g: self,
                     e_meta,
                 },
             )),
             Direction::BOTH => Box::new(itertools::chain!(
-                self.neighbours_window_t(v, r, Direction::IN),
-                self.neighbours_window_t(v, r, Direction::OUT)
+                self.edges_window_t(v, w, Direction::IN),
+                self.edges_window_t(v, w, Direction::OUT)
             )),
         }
     }
+
+    pub(crate) fn neighbours(
+        &self,
+        v: u64,
+        d: Direction,
+    ) -> Box<dyn Iterator<Item = VertexView> + Send + '_>
+    where
+        Self: Sized,
+    {
+        let edges = self.edges(v, d);
+
+        Box::new(edges.map(move |edge| {
+            let EdgeView { src_id, dst_id, .. } = edge;
+
+            let src_g_id = edge.src_g_id;
+            let dst_g_id = edge.dst_g_id;
+
+            if v == src_g_id {
+                VertexView::new(dst_g_id, dst_id)
+            } else {
+                VertexView::new(src_g_id, src_id)
+            }
+        }))
+    }
+
+    pub(crate) fn neighbours_window(
+        &self,
+        v: u64,
+        w: &Range<i64>,
+        d: Direction,
+    ) -> Box<dyn Iterator<Item = VertexView> + Send + '_>
+    where
+        Self: Sized,
+    {
+        let edges = self.edges_window(v, w, d);
+
+        Box::new(edges.map(move |edge| {
+            let EdgeView { src_id, dst_id, .. } = edge;
+
+            let src_g_id = edge.src_g_id;
+            let dst_g_id = edge.dst_g_id;
+
+            if v == src_g_id {
+                VertexView::new(dst_g_id, dst_id)
+            } else {
+                VertexView::new(src_g_id, src_id)
+            }
+        }))
+    }
+
+    pub(crate) fn neighbours_ids(
```

Comment:
Do we actually use any of these global (non-windowed functions anywhere. I.e. would it be easier to have everything have a window and remove?